### PR TITLE
Fix Debug build break with device_uvectors in grouped_rolling.cu

### PR DESCRIPTION
**Pull request by mythrocks**
After #7523, on `Debug` builds, one seems to see the following build error:
```
cudf/cpp/src/rolling/grouped_rolling.cu(130): error: no operator "[]" matches these operands
            operand types are: const cudf::groupby::detail::sort::sort_groupby_helper::index_vector [ int ]
...
cudf/cpp/src/rolling/grouped_rolling.cu(130): error: no operator "[]" matches these operands
            operand types are: const cudf::groupby::detail::sort::sort_groupby_helper::index_vector [ unsigned long ] 
```

The offending line has an `assert()` that is only compiled during `Debug` builds. This commit corrects the indexing into `device_uvector`.

### CODE REVIEW

**Comment by mythrocks** [comment]
Alternatively, we might just remove the `assert()`.

File: cpp/src/rolling/grouped_rolling.cu [hunk 1]
```diff
@@ -127,8 +127,9 @@ std::unique_ptr<column> grouped_rolling_window(table_view const& group_keys,
   //   groups.)
   //   3. [0, 500, 1000] indicates two equal-sized groups: [0,500), and [500,1000).
 
-  assert(group_offsets.size() >= 2 && group_offsets[0] == 0 &&
-         group_offsets[group_offsets.size() - 1] == input.size() &&
+  assert(group_offsets.size() >= 2 && group_offsets.element(0, rmm::cuda_stream_default) == 0 &&
+         group_offsets.element(group_offsets.size() - 1, rmm::cuda_stream_default) ==
+           input.size() &&
```
Comment (davidwendt) [inline cpp/src/rolling/grouped_rolling.cu:132]: Why not use the input `stream` parameter here?

**Review by devavret — CHANGES_REQUESTED** [review]
Not a part of this PR but can you please add the stream to the `group_offsets()` and `group_labels()`

File: cpp/src/rolling/grouped_rolling.cu (hunk 1 above)
Comment (mythrocks) [reply cpp/src/rolling/grouped_rolling.cu:132]: Hmm... I could have sworn there wasn't a `stream` parameter available at this point. Will fix this shortly.

**Comment by mythrocks** [comment]
> Not a part of this PR but can you please add the `stream` to the `group_offsets()` and `group_labels()`

Done. Thanks for pointing that out. TIL.

**Review by davidwendt — APPROVED** [review]

**Review by ttnghia — APPROVED** [review]

**Review by devavret — APPROVED** [review]

**Review by karthikeyann — APPROVED** [review]

**Comment by davidwendt** [comment]
@mythrocks Is this good to merge?

**Comment by mythrocks** [comment]
Yikes. Yes, this should be merged.

**Comment by mythrocks** [comment]
Rebased and retested. All tests are passing. I'll merge this now.

**Comment by mythrocks** [comment]
@gpucibot merge

**Pull request merged**